### ISSUE TITLE
HTMLElement.rootElement is deprecated.

### ISSUE THREAD

**Issue by nibushibu** [tony612/highlight-column#37]
The contents of `atom-text-editor` elements are no longer encapsulated
within a shadow DOM boundary. Please, stop using `rootElement` and access
the editor contents directly instead.
```
HTMLElement.rootElement (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-element.js:23:10)
HTMLDivElement.attachToLines (/Users/nibushibu/.atom/packages/highlight-column/lib/highlight-column-element.coffee:28:42)
HTMLDivElement.initialize (/Users/nibushibu/.atom/packages/highlight-column/lib/highlight-column-element.coffee:20:12)
<unknown> (/Users/nibushibu/.atom/packages/highlight-column/lib/main.coffee:34:74)
TextEditor.observeCursors (<embedded>:2994:15)
<unknown> (/Users/nibushibu/.atom/packages/highlight-column/lib/main.coffee:30:23)
```

**Comment by bennypowers** [comment]
Reproduction:

1. Upgrade from 1.18-beta3 to 1.19-beta0
2. restart atom to install

**Atom**: 1.19.0-beta0 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.2
**Thrown From**: [highlight-column](https://github.com/tony612/highlight-column) package 0.5.2


### Stack Trace

Failed to activate the highlight-column package

```
At Cannot read property 'NaN' of undefined

TypeError: Cannot read property 'NaN' of undefined
    at TextEditorComponent.renderedScreenLineForRow (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:900:37)
    at TextEditorComponent.pixelLeftForRowAndColumn (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:2218:35)
    at TextEditorComponent.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-component.js:191:46)
    at HTMLElement.pixelPositionForScreenPosition (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-element.js:243:32)
    at HTMLElement.pixelRectForScreenRange (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor-element.js:254:22)
    at HTMLDivElement.HighlightColumnView._cursorPixelRect (/packages/highlight-column/lib/highlight-column-element.coffee:85:27)
    at HTMLDivElement.HighlightColumnView.highlightRect (/packages/highlight-column/lib/highlight-column-element.coffee:75:13)
    at HTMLDivElement.HighlightColumnView.updateHighlight (/packages/highlight-column/lib/highlight-column-element.coffee:54:15)
    at /packages/highlight-column/lib/highlight-column-element.coffee:45:35
    at Config.module.exports.Config.observeKeyPath (/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:625:13)
    at Config.module.exports.Config.observe (/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:150:27)
    at HTMLDivElement.HighlightColumnView.handleConfigEvents (/packages/highlight-column/lib/highlight-column-element.coffee:47:35)
    at HTMLDivElement.HighlightColumnView.handleEvents (/packages/highlight-column/lib/highlight-column-element.coffee:19:28)
    at HTMLDivElement.HighlightColumnView.initialize (/packages/highlight-column/lib/highlight-column-element.coffee:7:6)
    at /packages/highlight-column/lib/main.coffee:25:63
    at TextEditor.module.exports.TextEditor.observeCursors (/Applications/Atom Beta.app/Contents/Resources/app/src/text-editor.js:744:15)
    at /packages/highlight-column/lib/main.coffee:23:14
    at Workspace.observeTextEditors (/Applications/Atom Beta.app/Contents/Resources/app/src/workspace.js:717:15)
    at Object.activate (/packages/highlight-column/lib/main.coffee:22:20)
    at Package.module.exports.Package.activateNow (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:253:25)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package.js:225:38
    at Package.module.exports.Package.measure (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:99:21)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package.js:218:32
    at Package.module.exports.Package.activate (/Applications/Atom Beta.app/Contents/Resources/app/src/package.js:215:40)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:645:40)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:626:35
    at Config.module.exports.Config.transactAsync (/Applications/Atom Beta.app/Contents/Resources/app/src/config.js:346:24)
    at PackageManager.module.exports.PackageManager.activatePackages (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:621:25)
    at PackageManager.module.exports.PackageManager.activate (/Applications/Atom Beta.app/Contents/Resources/app/src/package-manager.js:603:52)
    at /Applications/Atom Beta.app/Contents/Resources/app/src/atom-environment.js:843:36
```

### Commands

```
     -0:31.9.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-when-scrolling)
```

### Non-Core Packages

```
atom-beautify 0.30.0 
atom-material-syntax 1.0.6 
atom-material-syntax-dark 1.0.0 
atom-material-ui 2.0.4 
atom-pair 2.0.10 
atom-ternjs 0.18.3 
atomic-chrome 0.3.0 
auto-update-packages 1.0.1 
autocomplete-paths 1.0.5 
autocomplete-python 1.8.64 
busy-signal 1.4.3 
dash 1.7.1 
docblockr 0.10.3 
emmet 2.4.3 
emmet-snippets-compatibility 1.0.6 
file-icons 2.1.7 
git-log 0.4.1 
highlight-column 0.5.2 
highlight-line 0.12.0 
highlight-selected 0.13.1 
Hydrogen 1.17.1 
indent-guide-improved 1.4.12 
intentions 1.1.2 
kite 0.27.0 
language-fish-shell 1.0.5 
less-than-slash 0.16.0 
lines 0.13.1 
linter 2.2.0 
linter-csslint 1.3.4 
linter-docker 0.2.0 
linter-eslint 8.2.1 
linter-flake8 2.2.1 
linter-htmlhint 1.3.3 
linter-jshint 3.1.2 
linter-jsonlint 1.3.0 
linter-php 1.3.2 
linter-proselint 3.2.1 
linter-pycodestyle 2.0.2 
linter-sass-lint 1.7.4 
linter-stylelint 3.10.0 
linter-tslint 1.6.0 
linter-ui-default 1.6.1 
linter-write-good 0.9.0 
minimap 4.28.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-highlight-selected 4.6.1 
minimap-linter 2.0.0 
minimap-pigments 0.2.2 
multi-cursor 2.1.5 
pigments 0.39.1 
polymer-ide 0.3.0 
polymer-snippets 3.2.1 
project-manager 3.3.5 
regex-railroad-diagram 0.17.0 
relative-numbers 0.8.0 
sequential-number 0.5.0 
sync-settings 0.8.2 
termination 0.5.1 
todo-show 1.11.0 
toggle-quotes 1.0.1 
vim-mode-plus 0.93.0 
vim-mode-plus-keymaps-for-surround 0.2.1 
vim-plus-autocomplete-emmet-compatibility 0.1.0 
```

**Comment by tony612** [comment]
The deprecation warning is fixed in 0.5.3. The bug in 1.19-beta0 is tracked in #38
